src/dbnode/storage/shard.go

```diff
@@ -974,6 +962,62 @@ func (s *dbShard) writeAndIndex(
 	return series, wasWritten, nil
 }
 
+func (s *dbShard) SeriesReadWriteRef(
+	id ident.ID,
+	tags ident.TagIterator,
+	opts ShardSeriesReadWriteRefOptions,
+) (SeriesReadWriteRef, error) {
+	// Try retrieve existing series.
+	entry, shardOpts, err := s.tryRetrieveWritableSeries(id)
+	if err != nil {
+		return SeriesReadWriteRef{}, err
+	}
+
+	if entry != nil {
+		// The read/write ref is already incremented.
+		return SeriesReadWriteRef{
+			Series:              entry.Series,
+			UniqueIndex:         entry.Index,
+			ReleaseReadWriteRef: entry,
+		}, nil
+	}
+
+	// Avoid double lookup by enqueueing insert immediately.
```

Comment:
Don't think this is the correct comment here?

Reply:
True, removing.